### PR TITLE
feat(docs): add React Native tutorial

### PR DESCRIPTION
**Pull request by beeman**
This is a first go at the React Native tutorial for Solana. 

Please give me feedback on the content or tone of voice, happy to tweak things where needed!

### CODE REVIEW

File: docs/integrations/react-native.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+### Install dependencies
+
+Next, we install the dependencies. We install the Solana SDK, and in addition we install a package to patch the `metro` configuration, and two polyfills that patch the React Native environment. 
+
+```shell
+yarn add @solana/web3.js metro-config react-native-get-random-values react-native-url-polyfill
+```
+
+### Update `index.js`
```
Comment (jacobcreech) [inline docs/integrations/react-native.md:66]: Any explanation as to why polyfills and metro need to be updated? Would be a good fyi walking through this

File: docs/integrations/react-native.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    resolver: {
+      sourceExts: [...sourceExts, 'cjs', 'svg'],
+    },
+  };
+};
+```
+
+### Update `App.tsx`
+
+In this step, we can use the Solana SDK.
```
Comment (jacobcreech) [inline docs/integrations/react-native.md:107]:
```suggestion
Let's add a web3.js example into our app!
```

File: docs/integrations/react-native.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+Open the file `App.tsx` and add the following code inside the `App` function:
```
Comment (jacobcreech) [inline docs/integrations/react-native.md:109]: Maybe an explanation of what the example does here as well?

**Review by jacobcreech — CHANGES_REQUESTED** [review]
Some change requested here and there. Looks pretty good though. You should add a link to the starter project at the end once you have that built out

**Comment by jacobcreech** [comment]
🤔 Still thinking of how to properly format this in `integrations`. I may separate into 2 sections, one for CPI info and another for client integrations and put this in the latter. That'll be later though!

File: docs/integrations/react-native.md (hunk 1 above)
Comment (beeman) [reply docs/integrations/react-native.md:66]: The `Common issues` section at the end of the doc explains the 'why' in more detail, is that what you're aiming at?

**Comment by beeman** [comment]
> Some change requested here and there. Looks pretty good though. You should add a link to the starter project at the end once you have that built out

Thanks a lot for the feedback @jacobcreech!

I think I resolved most of the issues, feel free to give it another review! Thanks!

File: docs/integrations/react-native.md (hunk 1 above)
Comment (jacobcreech) [reply docs/integrations/react-native.md:68]: Good enough for now. Thanks!

**Review by jacobcreech — APPROVED** [review]
Thanks for this contribution!

**Pull request merged**